### PR TITLE
Add a media handler to respond to intra frame requests

### PR DESCRIPTION
**Pull request by kuzux**
This PR introduces a PLIResponder class, which can be used for responding back with an intra frame after the receiver requests one with an FIR (Full Intra Request) or a PLI (Picture Loss Indicator) message.

This is useful when a video is being streamed live and we can control the behavior of the video encoder. PLIResponder simply notifies its caller when a new intra frame is requested, which passes this request onto its video frame source.

I actually thought about contributing the generic PLIResponder thing we used upstream all the way back last year, but the implementation back then had some issues (later discovered that they were due to browsers grouping control messages)

Now, there are a couple things missing:
* The naming is temporary, since this responds to FIR messages as well as PLI
* I'm not sure about how to integrate this into the C API
* Also I'm not sure about how to integrate this into an example (This doesn't make sense in the existing examples since none of them can meaningfully respond to an intra frame request)

### CODE REVIEW

**Comment by kuzux** [comment]
Also noticed that #929 will definitely change things in relation to this :D

File: include/rtc/pliresponder.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "mediahandlerelement.hpp"
+#include <functional>
+
+namespace rtc {
+
+/// Responds to PLI and FIR messages sent by the receiver. The sender should respond to these
+/// messages by sending an intra. 
+class RTC_CPP_EXPORT PliResponder final : public MediaHandlerElement {
+    std::function<void(void)> onPli;
```
Comment (paullouisageneau) [inline include/rtc/pliresponder.hpp:22]: `onPli` should be a `Synchronized_callback` for thread safety. Also, private member variables should be prefixed by `m` in the library so the name should be `mOnPli`.

File: include/rtc/pliresponder.hpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#if RTC_ENABLE_MEDIA
+
+#include "mediahandlerelement.hpp"
+#include <functional>
+
+namespace rtc {
+
+/// Responds to PLI and FIR messages sent by the receiver. The sender should respond to these
+/// messages by sending an intra. 
+class RTC_CPP_EXPORT PliResponder final : public MediaHandlerElement {
```
Comment (paullouisageneau) [inline include/rtc/pliresponder.hpp:21]: I'm not a fan of the naming, `Pli` is fine by me but this media handler does not really respond to anything (meaning it doesn't send messages). You could call it something more generic like `PliHandler`.

File: include/rtc/plihandler.hpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#include "mediahandlerelement.hpp"
+#include "utils.hpp"
+#include <functional>
+
+namespace rtc {
+
+/// Responds to PLI and FIR messages sent by the receiver. The sender should respond to these
+/// messages by sending an intra. 
+class RTC_CPP_EXPORT PliHandler final : public MediaHandlerElement {
+    std::function<void(void)> mOnPli;
```
Comment (paullouisageneau) [inline include/rtc/plihandler.hpp:23]: This should be changed to `synchronized_callback` as it's the actual callback being called by this class.

File: src/plihandler.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+				mOnPli();
+				break;
+			}
+		}
+		offset += header->lengthInBytes();
+	}
+	return { message, std::nullopt };
+}
+
+PliHandler::PliHandler(rtc::synchronized_callback<> onPli) : mOnPli(onPli) { }
```
Comment (paullouisageneau) [inline src/plihandler.cpp:40]: This should stay a `function`.

**Review by paullouisageneau — APPROVED** [review]
Looking good, thanks.

**Pull request merged**